Co dump button

daschuer opened the pull request:
This adds a Dump button above the CO table in developer menu. 
This could be very helpful to nail down user issues and do manual tests.

uklotzde committed:
Merge branch 'NewSoundAPI' of https://github.com/illuusio/mixxx.git into NewSoundSourceAPI

uklotzde committed:
SoundSourceMediaFoundation: Fix compile errors on Windows

uklotzde committed:
SoundSourceMediaFoundation: Fix #frames to #samples conversion

Mathematically equivalent, but now the intention should be clearer.

uklotzde committed:
SoundSourceMediaFoundation: Improve readability of bitrate calculation

uklotzde committed:
SoundSourceMediaFoundation: Use SINT instead of int/long

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

demos committed:
Merge pull request #1 from daschuer/pan_effect_stereo

Introduced a EngineFilterPan class
Thank you very much, i'll try to integrate it fastly :)

uklotzde committed:
Add missing files to build script of SoundSourceMediaFoundation plugin

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

uklotzde committed:
Fix decoding of some "flaky" FLAC files

Thanks to Stephan Balmer for the detailed report and the files!

uklotzde committed:
Avoid unnecessary seeking in FLAC files

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

uklotzde committed:
Improve decoding of corrupt/malformed FLAC files

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

uklotzde committed:
Change initialization order in SoundSourceMediaFoundation plugin

...and fix the "bits per sample" attribute

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

Conflicts:
	src/library/browse/browsetablemodel.cpp

uklotzde committed:
Merge branch 'master' into NewSoundSourceAPI

uklotzde committed:
Provide hints for opening audio sources via AudioSourceConfig

jercaianu committed:
[Bug #1299029] phaser effect

rryan committed:
Fix the build on OS X.

rryan committed:
Merge pull request #411 from uklotzde/NewSoundSourceAPI

New SoundSource/AudioSource API

ywwg committed:
LateNight: Fix effect knobs cut off on Mac

uklotzde committed:
Fix the Windows build (SoundSourceMediaFoundation)

jercaianu committed:
Redone implementation with better sound output

jercaianu committed:
Changed depth limits to 0 and 1

jercaianu committed:
improved sound quality

rryan committed:
Merge pull request #562 from uklotzde/SoundSourceMediaFoundationBuild

Fix Windows build (SoundSourceMediaFoundation)

daschuer committed:
mp3: Workaround for different sample rates in on file

daschuer committed:
mp3: made derPerFrameRate local

daschuer committed:
mp3: move switch case blocks to helper functions

rryan committed:
Add "stabilization read" to seeks for MP3s in SoundSourceCoreAudio.

SoundSourceProxyTest.seekForward was failing on OSX due to
cover-test.mp3 not producing accurate samples on a seek. It seems this
is related to the MP3 decoder not being "primed" right after a seek. To
compensate for this, we seek backwards 29 MP3 frames (same as in
SoundSourceMp3) and read forward.

The test passes now but it would be ideal if we had a reliable way to
determine the right amount to read on a per-file basis from ExtAudioFile
directly (in case other formats need pre-fetching). I'm following up
with Apple support for help here.

daschuer committed:
Merge pull request #564 from daschuer/mp3samplerate

mp3: Workaround for different sample rates in on file

rryan committed:
Merge pull request #565 from rryan/extaudiofile

Add "stabilization read" to MP3 seeks in SoundSourceCoreAudio.

jercaianu committed:
Rewrote phaser with much better sound output

uklotzde committed:
Fix handling of mono MP3 frames within a stereo file

uklotzde committed:
Abort loading of MP3 file on error before determining sampling rate

uklotzde committed:
Limit the maximum number of channels in MP3 files to 2

uklotzde committed:
Fix debug assertion to accept MP3 files without any tags

uklotzde committed:
Some minor code reorderings when loading MP3 files

uklotzde committed:
Fix libMAD error handling for MP3 decoding

The decoding functions do NOT return the actual error code!! Instead
the error code must be read from mad_stream if the decoding functions
returned != 0.

This caused some really ugly sounds at the beginning of some MP3 files,
because frames were not skipped as suppossed to be.

uklotzde committed:
Separate code paths for reading mono/stereo MP3 data

Slightly more code, but hopefully easier to understand.

daschuer committed:
Merge pull request #568 from uklotzde/MP3ChannelAndDecodingFix

MP3 channel and decoding fix

uklotzde committed:
Fix Coverity warning for SoundSourceSndFile

*** CID 61970:  Uninitialized members  (UNINIT_CTOR)
/src/sources/soundsourcesndfile.cpp: 17 in Mixxx::SoundSourceSndFile::SoundSourceSndFile(QUrl)()
11         return list;
12     }
13
14     SoundSourceSndFile::SoundSourceSndFile(QUrl url)
15             : SoundSource(url),
16               m_pSndFile(NULL) {
>>>     CID 61970:  Uninitialized members  (UNINIT_CTOR)
>>>     Non-static class member field "m_sfInfo.seekable" is not initialized in this constructor nor in any functions that it calls.
17     }
18
19     SoundSourceSndFile::~SoundSourceSndFile() {
20         close();
21     }

uklotzde committed:
Fix Coverity warning for SoundSourceMp3

*** CID 61968:  Uninitialized members  (UNINIT_CTOR)
/src/sources/soundsourcemp3.cpp: 163 in Mixxx::SoundSourceMp3::SoundSourceMp3(QUrl)()
157               m_fileSize(0),
158               m_pFileData(NULL),
159               m_avgSeekFrameCount(0),
160               m_curFrameIndex(kFrameIndexMin),
161               m_madSynthCount(0) {
162         m_seekFrameList.reserve(kSeekFrameListCapacity);
>>>     CID 61968:  Uninitialized members  (UNINIT_CTOR)
>>>     Non-static class member field "m_madSynth.pcm" is not initialized in this constructor nor in any functions that it calls.
163     }
164
165     SoundSourceMp3::~SoundSourceMp3() {
166         close();
167     }

uklotzde committed:
Eliminate unneeded member variable in SoundSourceSndFile

uklotzde committed:
Symmetric invocation of init()/finish() functions in SoundSourceMp3

jercaianu committed:
Added stereo option and removed redundant coefficient computing

rryan committed:
Merge pull request #569 from uklotzde/FixCoverityWarningsForSoundSources

Fix coverity warnings for new SoundSources

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Do not segfault when writing tags into files with inaccessible tags

Observed for some .m4a files when TagLib failed to access (read/write) the
existing MP4 tag.

demos committed:
missing coef

demos committed:
Merge branch 'pan_effect_stereo' of https://github.com/demos/mixxx into pan_effect_stereo

jercaianu committed:
Phaser now has only even number of stages

uklotzde committed:
Improve readability of tag writing code

daschuer committed:
Merge pull request #571 from uklotzde/TagWritingFixes

Writing Tags: Handle inaccessible tags correctly

demos committed:
beginning of period unit switch

jercaianu committed:
Tweaked some default values and added better names for variables

jercaianu committed:
Reformated sample processing part

default-kramer committed:
fix Windows build

rryan committed:
Version bump master to avoid confusion with 1.12 branch.

rryan committed:
Merge pull request #572 from default-kramer/fixWindowsBuild

fix Windows build

jercaianu committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into phaser

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

demos committed:
the delay moves with the sound

demos committed:
cleaning

daschuer committed:
Coverity Scan: Removed dead code

daschuer committed:
Fix error in CachingReader::read that prevent seeking into preroll, fixes Bug #1450260

demos committed:
cleaning + better texts + delay debug

uklotzde committed:
Rename min/max frame index functions to improve readability

uklotzde committed:
Cleanup AudioSource API and use consistent naming

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Add note about SIGBUS errors if MP3 file becomes inaccessible

https://bugs.launchpad.net/mixxx/+bug/1452005

demos committed:
changing the scale of the period knob

demos committed:
changing the scale of the period knob

demos committed:
logarithmic smoothing + cleaning

uklotzde committed:
Avoid unnecessary integer casts

uklotzde committed:
Replace "magic numbers" with constants

uklotzde committed:
Reduce visibility of constants and functions in AudioSource

uklotzde committed:
Move MP3 specific constant into separate header file

uklotzde committed:
Remove redundant static qualifier from constant in anonymous namespace

rryan committed:
Merge pull request #577 from uklotzde/AudioSourceApiCleanup

AudioSource API Code Cleanup

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Log warning about unreadable tags in audio files

uklotzde committed:
Change parameter type of getSoundSource() from QString to QUrl

daschuer committed:
enginefilterpane: use linear stetching instead of ramping

daschuer committed:
add enginefilterpanesingle

daschuer committed:
Used enginefilterpanesingle in auto pane effect

daschuer committed:
Use ramped value for delay

jercaianu committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into phaser

daschuer committed:
Added Dump button to the developers tools

daschuer committed:
Switch to csv dump

daschuer committed:
Add timestamp to co_dump file name

daschuer committed:
Added a co_dump from a inital Mixxx run

daschuer committed:
Added tooltip

ywwg commented:
target to master please, not 1.12

daschuer commented:
This is a useful feature for 1.12. It is very helpful for remote debugging.
Any source of information can help a hot to find bugs fast.

ywwg commented:
No new features in release branch 1.12.  We have to work with what we have for now, sorry.  This will be very useful work in the next version, please target to master.

ywwg committed:
Merge branch '1.12'

Conflicts:
	src/test/mixxxtest.cpp

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

uklotzde committed:
Add an assertion to detect invalid conversions QString <-> QUrl

uklotzde committed:
New streamlined SoundSource plugin API version 8

A simple C++ API that reduces the chance of memory leaks that were
imminent in the previous C/C++ version. Error handling and logging
in SoundSourceProxy while loading plugins has been improved.

uklotzde committed:
Allow CachingReaderWorker to return partially read chunks

This might happen for corrupt/truncated files. Currently there is no
special case handling in CachingReader. Partially (= incomplete) read
chunks are simply treated like successfully (= complete) read chunks.

Without this fix the CachingReader tried to read the remaining samples
in the last chunk of a corrupt MP3 file over and over.

daschuer commented:
This is hidden for the normal users and the scope of the changes is very limited. 
It feels stupid not to have this feature if we face a hard to debug problem. 
I see a bunch of reasons for this dump button but not a single reason against it. 
.. except your principles. Or are there additional once?

uklotzde committed:
Detach dynamic library management from SoundSourceProxy

ywwg commented:
I am trying to enforce a development pattern by which release branches are strictly managed to decrease chance of regression and increase velocity.  This post has a summary of some of the principles I'm applying: http://nvie.com/posts/a-successful-git-branching-model/  Note that we don't do everything like this model, but the bit about release branches applies:

"bug fixes may be applied in this branch (rather than on the develop branch). Adding large new features here is strictly prohibited. They must be merged into develop, and therefore, wait for the next big release."

By having a more-strict policy for release branches, we'll increase the speed of the current release.  

Ultimately, yes, this is about my principles which I've applied to great success in my professional life as evidenced by my years of tight-deadline project management.

jercaianu committed:
Fixed bubbling noise from the 'Rate' knob, added tanh clipper and comments

uklotzde committed:
Introduce SoundSourceProvider interface

SoundSourceProvider is a generic interface that will implemented both by
plugin libraries as well as built-in sound sources. This will simplify
the selection of an appropriate SoundSource in SoundSourceProxy and is
useful for future extensions like selecting one out of multiple providers
for each file type.

uklotzde committed:
Add and register a SoundSourceProvider for each SoundSource

uklotzde committed:
Fix FFmpeg build

uklotzde committed:
Give SoundSource providers a name

jercaianu committed:
added tanh

daschuer commented:
Since this is a not a feature, for the user but a debug tool my hope is that this help so speed up the release process and will speed up finding bugs. 
So it should fit to the goals you are trying to reach with your principals.
Is there an argument against it? 

What do the other think?

uklotzde committed:
Delete obsolete #include directive

jercaianu committed:
clipped only feedback

jercaianu committed:
updated frequency description

uklotzde committed:
Wrap static members from SoundSourceProxy into class

uklotzde committed:
Fix static initialization of SoundSource providers in tests

uklotzde committed:
Fix SoundSource plugin warning

uklotzde committed:
Log detailed debug info about SoundSource providers

uklotzde committed:
Move final debug logging out of loop

uklotzde committed:
Delete obsolete member from SoundSourceProxy

uklotzde committed:
Avoid unnecessary locking in SoundSourceProxy

uklotzde committed:
Eliminate duplicate initialization in tests

uklotzde committed:
Add debug logging for seek test

jercaianu committed:
Fixed comment indentation and rate knob issue

jercaianu committed:
Added linear knob until toggle one is fixed

uklotzde committed:
Simplify SoundSource plugin registration

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

demos committed:
adding comments to filterpansingle

Commit:
Changed RateRange to RateRangeDecimal

demos committed:
adding comments to filterpansingle 2

demos committed:
Merge pull request #2 from daschuer/pan_effect_stereo

pan delay per sample with linear stretching

demos committed:
Merge branch 'daschuer_filterpansingle_for_autopan' of https://github.com/demos/mixxx into pan_effect_stereo

demos committed:
Merge branch 'pan_effect_stereo' of https://github.com/demos/mixxx into pan_effect_stereo

daschuer committed:
Merge pull request #586 from uklotzde/CachingReaderPartialChunkRead

CachingReaderWorker: Don't discard partially read chunks

uklotzde committed:
Delete obsolete member functions and code cleanup

uklotzde committed:
Use plugin file name filter only for Apple OS X

At least this is how it works in 1.12.

Commit:
Keep ComboBoxRateRange and add some more sane values, plus allow any
setting in the config file

Commit:
Limit Rate to 90% max and better labels

Commit:
Surround * and / with spaces

Commit:
Name new config RateRangePercent and rename slot/variables accordingly

daschuer committed:
Merge pull request #560 from jercaianu/phaser

phaser effect

uklotzde committed:
Change parameter name: fileName -> libFilePath

uklotzde committed:
Change getter name: getFileName() -> getFilePath()

The corresponding Qt function name is actually misleading.

uklotzde committed:
Change the name of the MP3 SoundSource provider

uklotzde committed:
Add a TODO note about network shares

According to the documentation of QUrl::isLocalFile()/toLocalFile()
network shares are handled in the same way as before, i.e. they are
considered to be local files with all consequences.

uklotzde committed:
Filtering of plugin file names on UNIX systems (not only OS X)

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

demos committed:
pausable period

daschuer committed:
Merge pull request #588 from MK-42/rateAsDecimal

Changed RateRange to RateRangeDecimal

ywwg committed:
Minor style and English language fixes

demos committed:
position of the sound fixed when period knob is used

demos committed:
period will not be fixed anymore if the effect is in enabling state or the position change is ramped

demos committed:
period will not be fixed anymore if the effect is in enabling state or the position change is ramped

demos committed:
reintroducing lawcoef : *2 at center and * 1.7 at left or right

uklotzde committed:
Streamline the MetadataSource interface

Usually both metadata and cover art should be parsed at once. Optional
parameters are used if only one or the other is needed. A single
virtual function covers all those use cases.

uklotzde committed:
Delete obsolete member function from TrackDAO

uklotzde committed:
Add checks for supported file types to SoundSourceProxy

uklotzde committed:
Remove obsolete local variables

uklotzde committed:
Consistent naming: "file type" instead of "file extension"

uklotzde committed:
Delete obsolete functions from SoundSourceFFmpeg

daschuer committed:
Merge pull request #460 from demos/pan_effect_stereo

Pan effect

Commit:
Highlight Crates a Track is in. Still some work to do

Commit:
Rename delegate according to his function

uklotzde committed:
Fix indentation

uklotzde committed:
Fix and comment parsing of cover art

uklotzde committed:
Restore deleted comment

uklotzde committed:
Improve comment in MetadataSource interface

uklotzde committed:
Add and improve functions for backward compatibility

Commit:
Highlighting crates the selected track is in works now

Commit:
Some minor code-style fixes

Commit:
Added comment to CrateHighlightDelegate::initStyleOption

Commit:
Added highlighting of playlists

Commit:
Another aproach for Repainting preventing the selection to move

rryan committed:
Ignore QModelIndex's that don't have TreeItem internal pointers in CrateHighlightDelegate.

rryan committed:
Fix SidebarModel handling of child models' dataChanged signals.

MK-42 committed:
Merge pull request #1 from rryan/sidebar-highlight

Sidebar highlight fixes.

Commit:
Holding Track-Crate relation in memory for faster highlighting

Commit:
Holding Track-Playlist relation in memory for faster highlighting

Commit:
Clean up used memory

Commit:
fix missing signals

Commit:
Clear selected Playlists/Crates on viewChange

rryan committed:
Remove CrateHighlightDelegate.

rryan committed:
Add bold attribute to TreeItem.

rryan committed:
Restore LibraryFeature::trackSelected signal.

I think this was why we had to add signals to each individual feature.

rryan committed:
Respond to Qt::FontRole in SidebarModel.

rryan committed:
Populate playlist and crate TreeItems with bold attribute when track selection changes.

rryan committed:
Populate TreeItem state when re-building the child model.

This supports bolding/unbolding the playlist/crate when a track is added/removed.

MK-42 committed:
Merge pull request #2 from rryan/sidebar-highlight

Some fixes.

rryan committed:
Swap key/value pairing of crate/playlist membership cache in CrateDAO/PlaylistDAO.

Also add a method to fetch all the playlists/crates a track is in.

uklotzde committed:
Correctly distinguish between "file extension" and "SoundSource type"

rryan committed:
Fetch playlist / crate membership set on track selection.

Avoids repeatedly asking the DAO for pair-wise membership (requires
linear hash table lookups).

uklotzde committed:
Initialize FFmpeg before registering providers and loading plugins

uklotzde committed:
Delete redundant DEBUG_ASSERT

Empty file names are handled gracefully by SoundSourceProxy.

rryan committed:
Fix type mismatch.

rryan committed:
Add move constructor / move assignment operator for SampleBuffer.

Fixes the build for -std=c++0x.

rryan committed:
Enable -std=c++11 support.

rryan committed:
Update Travis to use gcc 4.8 for c++11 support.

uklotzde committed:
Restore comment about QApplication in tests

uklotzde committed:
Reorder and add missing #include directives

uklotzde committed:
Static allocation of SoundSourceProvider in plugins

uklotzde committed:
Symmetrically allocate and delete SoundSource from within plugins

uklotzde committed:
Use nullptr (C++11) instead instead of NULL (deprecated)

uklotzde committed:
Fix move constructor in SampleBuffer

rryan committed:
Merge pull request #604 from uklotzde/SampleBufferMoveConstructorFix

Fix move constructor of SampleBuffer

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

uklotzde committed:
Use C+11 override keyword

uklotzde committed:
Add comments about singleton/shared SoundSourceProvider instances

uklotzde committed:
Unify export of SoundSources from plugin libraries

uklotzde committed:
Prefer local static variables for shared data in plugins

rryan committed:
Merge pull request #578 from uklotzde/SoundSourcePluginUrlFileNameFix

SoundSource Plugin API: Version 8

rryan committed:
Fix c++11 TODO.

uklotzde committed:
Fix memory leak in NotNode

uklotzde committed:
Add (redundant) keyword 'virtual' to declaration of destructor

rryan committed:
Add public domain std::make_unique shim from Brandon Streiff.

Taken from: https://github.com/bstreiff/cppbits

Thank you @bstreiff!

uklotzde committed:
Replace plain pointers with std::unique_ptr to avoid memory leaks

uklotzde committed:
Delete obsolete constructors and destructors

uklotzde committed:
Add 'override' keyword to inherited virtual functions in derived classes

uklotzde committed:
Fix (or comment) fragile design of Numeric-/DurationFilterNode

Virtual member functions should not be called from within the constructor!

uklotzde committed:
Consistent wrapping of SQL expressions into parantheses

uklotzde committed:
Improve readability using auto and std::make_unique

uklotzde committed:
Don't use auto for derived class ptr to fix compile error in CI build

uklotzde committed:
Revert previous changes and try again to fix CI build errors

rryan committed:
Remove QColor::darker call.

Since it does not mutate its instance it has no effect.

rryan committed:
Remove no-args constructor parentheses.

rryan committed:
Use QDomNodeList::count instead of length.

In Qt5 QDomNodeList::length changed from unsigned int to int. To avoid
compiler warnings when using one or the other use count.

rryan committed:
Move QtScript ByteArrayClass into lib/.

rryan committed:
Move reverb plugin to lib/reverb.

rryan committed:
Remove some instances of FALSE and TRUE.

rryan committed:
Remove use of deprecated (in c++11) register keyword.

The generated assembly with and without is identical in EngineFilterBessel4/8.

rryan committed:
Fix crate/playlist membership fetchers.

rryan committed:
Don't call TreeItemModel::setData in a loop.

This caused severe performance issues since it triggered a repaint on every loop
iteration.

rryan committed:
Merge pull request #617 from rryan/sidebar-highlight

Highlight Crates a Track is in

rryan committed:
Merge pull request #614 from uklotzde/QueryNodeRefactoring

QueryNode: Memory leak fix + refactoring

uklotzde committed:
CachingReader: Always read complete chunks

Fixes the following assertion:
DEBUG ASSERT: "0 == (hint.sample % CachingReaderWorker::kChunkChannels)" in file src/cachingreader.cpp, line 492

The calculation of chunk offsets relies on chunks with a constant size.
Only the last chunk might be smaller than all others. The worker now fills
the chunks with silence if the data has been read partially.

uklotzde committed:
CachingŔeaderWorker: Don't seek beyond EOF

Because of reasons I don't know processChunkReadRequest() is called
periodically even while the track is not playing. Unnecessary seek
operations on the SoundSource should be avoided for if the frame index
is out of range or beyond the end of the file.

uklotzde committed:
Handle trivial/edge cases in SoundSourceMp3 separately to avoid prefetching

uklotzde committed:
Align seeking in SoundSourceM4A with SoundSourceMp3 to avoid prefetching

uklotzde committed:
Merge branch '1.12' of https://github.com/mixxxdj/mixxx.git into Merge1.12ToMaster

# Conflicts:
#	src/library/searchquery.h
#	src/library/searchquery.cpp
#	src/main.cpp
#	src/soundsourcemodplug.cpp
#	src/test/coverartcache_test.cpp
#	src/test/soundproxy_test.cpp

daschuer committed:
Merge pull request #628 from uklotzde/Merge1.12ToMaster

Resolve merge conflicts from 1.12 to master

uklotzde committed:
Merge branch 'master' into CachingReaderAvoidUnnecessarySeeking

uklotzde committed:
Improve error dectection when seeking corrupt MP3/M4A files

uklotzde committed:
Fix wrong assert in SoundSourceM4A

uklotzde committed:
No need to overwrite the result value from the MP3 decoder

daschuer committed:
Code style and comments

daschuer committed:
Merge remote-tracking branch 'upstream/master' into scaler

daschuer committed:
ScratchTestmaster reflects now the ramping during Scratching

daschuer committed:
Merge remote-tracking branch 'upstream/1.12'

uklotzde committed:
Replace NULL with nullptr

uklotzde committed:
Merge branch 'master' into CachingReaderAvoidUnnecessarySeeking

uklotzde committed:
Add missing nullptr check

daschuer committed:
Merge pull request #626 from uklotzde/CachingReaderAvoidUnnecessarySeeking

Fix outstanding SoundSource issues

daschuer committed:
Merge pull request #599 from Frydac/bug_1316785

Bug 1316785 -- improve controller preset directory buttons

daschuer committed:
Merge branch '1.12'

daschuer commented:
Any additional comments?

ywwg commented:
I strongly prefer this be applied to master, not 1.12.  Obviously I can't force you to change your mind, but I've stated my objections clearly.  I've gotten tired of fighting you on these tiny issues and have mostly given up.

daschuer committed:
Merge pull request #546 from daschuer/scaler

Scaler fixes 4

esbrandt commented on src/dlgdevelopertoolsdlg.ui:
```diff
@@ -40,7 +40,17 @@
+       <item row="0" column="1">
+        <widget class="QPushButton" name="controlDump">
+         <property name="toolTip">
+          <string>Dumps all ControlObject values to a csv-file saved in the settings path (e.g. ~/.mixxx)</string>
```
This path in the example is only valid for Linux, might be better to link to a wiki/manual page were we list the actual path for all supported platforms.

esbrandt replied:
Arrrg, scratch that. Just noticed the text is in the tooltip, a link wont be clickable in there.

esbrandt replied:
I see that this feature is useful, but i`d prefer to merge this to master.
The line has already constantly been redrawn, otherwise we wont see 1.12 in 2015.

ywwg committed:
Merge branch '1.12'

daschuer commented:
Thank you for review. Since GitHub does not allow to retarget a PR I will commit it directly to master.
